sqlmesh/core/snapshot/definition.py

```diff
@@ -499,8 +502,16 @@ def merge_intervals(self, other: Snapshot) -> None:
         Args:
             other: The target snapshot to inherit intervals from.
         """
+        effective_from_ts = to_timestamp(self.effective_from) if self.effective_from else 0
+        apply_effective_from = effective_from_ts > 0 and self.fingerprint != other.fingerprint
+
         for start, end in other.intervals:
-            self.add_interval(start, end)
+            # If the effective_from timestamp is not None, then intervals that come after it must come from
```

Comment:
effective_from is never none, so this comment is misleading

Reply:
why is this section needed? conceptually i understand what you're trying to do, but we don't invalidate old intervals right, so what is this accomplishing

Reply:
Updated the comment